src/core/reactor.cc

```diff
@@ -4492,6 +4516,22 @@ void smp::configure(const smp_options& smp_opts, const reactor_options& reactor_
         .no_poll_aio = !reactor_opts.poll_aio.get_value() || (reactor_opts.poll_aio.defaulted() && reactor_opts.overprovisioned),
     };
 
+    // Disable hot polling if sched wakeup granularity is too high
+    // dio thread will be starved otherwise
+    // see https://github.com/scylladb/seastar/issues/2696
+    if (!reactor_cfg.no_poll_aio || reactor_cfg.max_poll_time != 0us) {
+        auto [wakeup_file, granularity] = wakeup_granularity();
```

Comment:
I don't think max_poll_time is relevant. If no_poll_aio is false (default) we enter the if and don't care about max_poll_time. If it's true, then we may still want to spin for a few microseconds before entering the kernel to avoid the expensive sleep path.

Reply:
I guess the spin still takes some time and I think the default is 100usec on a virtualized environment. Not sure what's best. Since it's a bad case, anyway, maybe setting it to zero is the safer step.

Note with tuned (not tuned(8)) setting of 0.5ms scheduling granularity and 100us spins,  we can spend a large fraction of time spinning.

Reply:
Yeah as per my tests in the issue thread changing `--idle-poll-time-us` definitely does have an additional impact on top of `--poll-aio` but yes the latter is the more important one.